TempleEight/docs/getting-started.md

```diff
@@ -4,3 +4,400 @@ title: Getting Started
+
+#### Auth Blocks
+
+The first piece of metadata in the `User` service is the `#auth` block. Having an auth block in a service means that an end user of the application can log into the system *as a* `User`, given their `email`. This defines a `User` as an identifiable entity in the service, meaning they can create other objects and hold ownership of them (i.e. A `User` could have `Pictures` in an image service that belong to them, and not other `Users`). 
+
+In terms of implementation, having an `auth` block anywhere in a Templefile causes the Temple CLI to generate an extra service, called `Auth`, which a user must authenticate themselves with to get a valid [`JWT`](https://jwt.io/) in order to access other services. This authentication comes in the form of an email and password login system.
+
+#### Permission Metadata
+
+Another type of metadata, present in every service are permission based metadata. These consist of `#readable`, `#writable`, and `#enumerable`. They respectfully define which other entities can read, write and enumerate the values stored in this service. 
+
+`#readable` has two options, `#readable(by: this)` is the default value when the service contains an auth block, and is `#readable(by: all)` when it doesn't. `by: this` defines that the only entity that can read the values stored in this entity, is this entity itself (i.e. The only `User` who can read `Bob's` date of birth, is `Bob` himself). This is opposed by `#readable(by: all)` which specifies that all `Users` can get information about all other users. 
+
+`#writeable` follows the same logic as `#readable`, except refers to the permission to write the data stored in an entity, rather than reading.
+
+`#enumerable` means that the entities in this service can be listed, by the entity that created them. It generates the `/list/{id}` endpoint on the generated microservice, which when invoked, will return all entities stored in the service, that hold `createdBy` value of `id`.
+
+
+```
+Match: service {
+  // Implicit `id` field
+  // Implicit `created_by` field, which will be Auth/User ID of creator
+  userOne: User;
+  userTwo: User;
+  matchedOn: datetime @serverSet;
+  #enumerable; // Adds list endpoint, filters by `created_by`, only creator can GET
+  // Implicit #readable(by: all), everyone can GET
+  // Implicit #writable(by: all), everyone can UPDATE and DELETE 
+}
+```
+
+
+#### CreatedBy Attributes
+
+As well as implicit `id` attributes, some services also get an implicit `createdBy` attribute generated. Those services are ones that do not contain an `#auth` block, but are contained in a Templefile where some service(s) do have an `#auth` block. An example of this can be seen in the second service block of the example `DatingApp` Templefile. The `Match` service stores an implicit `createdBy` attribute, which means that a `User` can enumerate all of their matches that have a `createdBy` attribute of their `id`. 
```

Comment:
This doesn’t sound deterministic enough. In a project where at least one service has an auth block, every service without its own auth block will have this attribute.